src/core/Entry.cpp

```diff
@@ -784,3 +791,29 @@ QString Entry::resolvePlaceholder(const QString& str) const
 
     return result;
 }
+
+QString Entry::resolveUrl(const QString& url) const
+{
+#ifdef WITH_XC_HTTP
+    QString newurl = url;
+    if (!url.contains("://")) {
+        // URL doesn't have a protocol, add https by default
+        newurl.prepend("https://");
```

Comment:
I'm not really sure if we should fallback to `https` instead of `http` when the url doesn't have a scheme. While it would be fine for most big sites like google or facebook, it could return a 404 on older sites that don't support SSL. I don't think this would be a security issue since we aren't really submitting any data to the site, just a request for the favicon, and in most cases the site can redirect the request to a `https` url, like google does:

![image](https://user-images.githubusercontent.com/107672/27762960-89fccc14-5e51-11e7-83f7-139d3553a86b.png)